main01.js

```diff
@@ -0,0 +1,39 @@
+let datum = document.querySelector(".datum");
+let searchbar = document.getElementById("searchbar");
+let dataStore = [];
+
+getdata();
+
+function getdata() {
+    fetch( 
+    'http://api.mediastack.com/v1/news?access_key=bff3a86eb00042ed68258f5c86a6318d'
+  )
+    .then((response) => {
+      const data = response.json();
+      return data;
+    })
+    .then((data) => {
+        dataStore = data.data;
+        datum.innerHTML = getHTML(data.data);
+    });
+}
+
+function getHTML(data){
+    return data.map(({title, url, source}) => generateHTML(title, url, source)).join('');
+}
+
+function generateHTML(title, url, source){
+    return `<div class="datapiece"><h1 class= "link"> <p> Link</p> <a> ${url} </a> </h1><h1 class= "link"> <p> Title </p> ${title}</h1><h1 class= "link"> <p> Source</p> ${source}</h1>
```

Comment:
Keep `datapiece` in the html and consider a better classname
`class= "link"` is not a valid attribute
`<a> ${url} </a>` needs to have an `href` attribute in there
you should probably split this up for readability
something like:
```
const link = `<h1 class= "link"> <p> Link</p> <a> ${url} </a> </h1>`; //
const title = <h1 class= "link"> <p> Title </p> ${title}</h1>; //  put ${title} in a tag
const source = <h1 class= "link"> <p> Source</p> ${source}</h1>;  //  put ${source} in a tag
return link + title + source;
```